packages/shared/src/utils/validator/validator.ts

```diff
@@ -90,8 +90,8 @@ class Validator {
                 const result = ruleObject.validator(this.input[attribute], ruleObject.options, this.store, this.input);
                 if (typeof result === 'boolean' && !result) {
                     this.addFailure(attribute, ruleObject);
-                } else {
-                    const { is_ok, message } = result as { is_ok: boolean; message: string };
+                } else if (typeof result === 'object') {
```

Comment:
If `result` is a `boolean` and `false`, then validation failed and we should make a call to `this.addFailure`. If `result` is a `boolean` and `true`, validation passed and nothing should be done. If `result` is an `object`, then we `destructure` it and check if `!is_ok`, meaning validation failed and we call `this.addFailure`, otherwise validation passed and nothing should be done